2011-08-03  Robert Millan  <rmh@gnu.org>

	* include/grub/zfs/zap_leaf.h (typedef union zap_leaf_chunk): Mark
	la_array as packed.
	Reported by: Zachary Bedell

--- a/include/grub/zfs/zap_leaf.h
+++ b/include/grub/zfs/zap_leaf.h
@@ -90,7 +90,7 @@
 		{
 			grub_uint8_t la_array[ZAP_LEAF_ARRAY_BYTES];
 			grub_uint64_t la_array64;
-		};
+		} __attribute__ ((packed));
 		grub_uint16_t la_next;		/* next blk or CHAIN_END */
 	} l_array;
 	struct zap_leaf_free {
